lib/words_report.rb

```diff
@@ -0,0 +1,25 @@
+class WordsReport
+  attr_reader :words
+
+  # Arbitrary definition of popular
+  POPULAR_COUNT = 5
+
+  def initialize(counts)
+    @counts = counts
+    @words = words_by_frequency
+  end
+
+  def popular_words(popular_count = POPULAR_COUNT)
+    @words.select { |word| @counts[word] >= popular_count }
+  end
+
+  def report(popular_count = POPULAR_COUNT)
+    popular_words(popular_count).each { |word| printf "%-12s: %3d\n", word, @counts[word] }
+  end
+
+  private
+
+  def words_by_frequency
+    @words = @counts.keys.sort { |a,b| [@counts[b], a] <=> [@counts[a], b] }
```

Comment:
Here's something else (not sure if better):

```ruby
@counts.sort_by { |word, count| [-count, word] }.map(&:first)
```

The `.map(&:first)` part of that is unfortunate.